distributed/shuffle/_shuffle.py

```diff
@@ -104,9 +104,15 @@ def rearrange_by_column_p2p(
     column: str,
     npartitions: int | None = None,
 ) -> DataFrame:
+    import pandas as pd
+
     from dask.dataframe.core import new_dd_object
 
     meta = df._meta
+    if not pd.api.types.is_integer_dtype(meta[column]):
```

Comment:
```suggestion
    if not pd.api.types.is_integer_dtype(meta[column].dtype):
```

Reply:
Works for me, is there any reason to picking one over the other?

Reply:
The first one is a pattern we want to get rid of at some point, these functions are too flexible